packages/beacon-node/src/network/core/networkCoreWorkerHandler.ts

```diff
@@ -130,21 +130,28 @@ export class WorkerNetworkCore implements INetworkCore {
       loggerOpts: modules.logger.toOpts(),
     };
 
-    const worker = new Worker(path.join(workerDir, "networkCoreWorker.js"), {
-      workerData,
-      /**
-       * maxYoungGenerationSizeMb defaults to 152mb through the cli option defaults.
-       * That default value was determined via https://github.com/ChainSafe/lodestar/issues/2115 and
-       * should be tuned further as needed.  If we update network code and see substantial
-       * difference in the quantity of garbage collected this should get updated.  A value that is
-       * too low will result in too much GC time and a value that is too high causes increased mark
-       * and sweep for some reason (which is much much slower than scavenge).  A marginally too high
-       * number causes detrimental slowdown from increased variable lookup time.  Empirical evidence
-       * showed that there is a pretty big window of "correct" values but we can always tune as
-       * necessary
-       */
-      resourceLimits: {maxYoungGenerationSizeMb: opts.maxYoungGenerationSizeMb},
-    } as ConstructorParameters<typeof Worker>[1]);
+    // Bun does not support resourceLimits
+    const workerOptions = (
+      "bun" in process.versions
```

Comment:
Should we use globalThis.Bun check instead?

Reply:
then this specific diff can be removed (it doesn't have anything to do with the internal package import pattern)

Reply:
Ok Removed.